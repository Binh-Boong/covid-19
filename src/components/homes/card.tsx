import { ICountry } from "../../types/apis/country.module";

export const Card = ({ 
  country, 
}: {
  country: ICountry,
}):JSX.Element => {
  return (
    <div className="w-full p-3 text-gray-700 shadow">
      <h3 className="mb-3 font-bold text-center">{country.Country}</h3>
      <div className="flex justify-between w-full mt-1">
        <div className="text-blue-800">New Confirmed</div>
        <div className="text-blue-800">{country.NewConfirmed}</div>
      </div>
      <div className="flex justify-between w-full mt-1">
        <div className="text-blue-800">Total Confirmed</div>
        <div className="text-blue-800">{country.TotalConfirmed}</div>
      </div>
      <div className="flex justify-between w-full mt-1">
        <div className="text-red-700">New Deaths</div>
        <div className="text-red-700">{country.NewDeaths}</div>
      </div>
      <div className="flex justify-between w-full mt-1">
        <div className="text-red-700">TotalDeaths</div>
        <div className="text-red-700">{country.TotalDeaths}</div>
      </div>
      <div className="flex justify-between w-full mt-1">
        <div className="text-green-700">New Recovered</div>
        <div className="text-green-700">{country.NewRecovered}</div>
      </div>
      <div className="flex justify-between w-full mt-1">
        <div className="text-green-700">Total Recovered</div>
        <div className="text-green-700">{country.TotalRecovered}</div>
      </div>
    </div>
  );
}