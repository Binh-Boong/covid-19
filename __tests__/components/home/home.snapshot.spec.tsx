import renderer from "react-test-renderer";

import { HomeComponent } from "../../../src/components/homes/home";

describe("Home Component", () => {
  it("renders correctly", () => {
      const card = renderer.create(
        <HomeComponent />
      ).toJSON();
    expect(card).toMatchSnapshot();
  });
});

export {}