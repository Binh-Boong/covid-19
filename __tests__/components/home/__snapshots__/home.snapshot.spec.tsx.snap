// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Card Component renders correctly 1`] = `
<div
  className="relative w-full min-h-screen"
>
  <div
    className="absolute top-0 bottom-0 left-0 right-0 z-50 flex flex-wrap content-center w-full h-full bg-gray-50"
  >
    <span
      className="relative block w-0 h-0 mx-auto my-0 text-old-navy-solid"
    >
      <div
        className="transform -translate-y-1/2"
      >
        <svg
          className="w-8 h-8 mr-3 animate-spin"
        >
          <use
            href="/icon.svg#loading"
          />
        </svg>
      </div>
    </span>
  </div>
  <div
    className="grid w-full grid-cols-1 gap-8 p-6 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 xl:grid-cols-5"
  />
</div>
`;

exports[`Home Component renders correctly 1`] = `
<div
  className="relative w-full min-h-screen"
>
  <div
    className="absolute top-0 bottom-0 left-0 right-0 z-50 flex flex-wrap content-center w-full h-full bg-gray-50"
  >
    <span
      className="relative block w-0 h-0 mx-auto my-0 text-old-navy-solid"
    >
      <div
        className="transform -translate-y-1/2"
      >
        <svg
          className="w-8 h-8 mr-3 animate-spin"
        >
          <use
            href="/icon.svg#loading"
          />
        </svg>
      </div>
    </span>
  </div>
  <div
    className="grid w-full grid-cols-1 gap-8 p-6 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 xl:grid-cols-5"
  />
</div>
`;
